### ISSUE TITLE
Infinite loop caused by `Layout/ClosingParenthesisIndentation`, `Layout/CommentIndentation`, `Layout/FirstArgumentIndentation` cops

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#14858]
## Steps to reproduce the problem

Run RuboCop with autocorrect on the following code:

```ruby
GroupUser
  # invited directly
  .where(user_id: topic_allowed_user_ids)
  .or(
    # invited via a group
  GroupUser.where(
      user_id: GroupUser.where(group_id: topic_allowed_group_ids).select(:user_id)
    )
  )
  .where(group_id: group.id)
  .select(:user_id)
  .distinct
  .count
```

## Expected behavior

It should not enter an infinite correction loop.

## Actual behavior

> Infinite loop detected in file.rb and caused by Layout/ClosingParenthesisIndentation, Layout/CommentIndentation, Layout/FirstArgumentIndentation -> Layout/ClosingParenthesisIndentation, Layout/CommentIndentation, Layout/FirstArgumentIndentation

--------

## RuboCop version

```
$ bundle exec rubocop -V
1.84.1 (using Parser 3.3.10.0, rubocop-ast 1.49.0, analyzing as Ruby 2.7, running on ruby 3.4.8) [x86_64-linux]
  - rubocop-performance 1.26.1
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.9.0
```
